subset-routing-test/src/main/java/com/hazelcast/jet/tests/client/subset/routing/SubsetRoutingTest.java

```diff
@@ -0,0 +1,202 @@
+                logger.info(logPrefix + "Validations count: " + validationCount);
+            }
+            sleepMinutes(sleepBeforeValidationInMinutes);
+
+        }
+        logger.info(logPrefix + "Final validations count: " + validationCount);
+    }
+
+    private void assertRequestToCluster(HazelcastClientInstanceImpl clientInstance) {
+        AtomicReference<Throwable> lastException = new AtomicReference<>();
+
+        ConditionVerifierWithTimeout verifier = new ConditionVerifierWithTimeout(
+                Duration.ofMinutes(maxWaitForExpectedConnectionInMinutes),
+                Duration.ofSeconds(sleepBetweenConnectionAssertionAttemptsInSeconds));
+
+        verifier
+                //This method throws OperationTimeoutException for unavailable cluster
+                .condition(() -> clientInstance.getMap(this.getClass().getSimpleName()).isEmpty())
+                .onException(lastException::set)
+                .onTimeout(() -> {
+                    if (lastException.get() != null) {
+                        throw new IllegalStateException("Timed out waiting for cluster active connection",
+                                lastException.get());
+                    }
+                    throw new IllegalStateException("Timed out waiting for cluster active connection");
+                })
+                .execute();
+
+    }
+
+    @Override
+    protected List<ClusterType> runOnClusters() {
+        return List.of(DYNAMIC, STABLE);
+    }
+
+    private void assertEffectiveMembersAndActiveConnections(HazelcastClientInstanceImpl clientInstance) {
+        Duration maxLoopDuration = Duration.ofMinutes(maxWaitForExpectedConnectionInMinutes);
+        Duration sleepBetweenAttempts = Duration.ofSeconds(sleepBetweenConnectionAssertionAttemptsInSeconds);
+
+        ConditionVerifierWithTimeout verifier = new ConditionVerifierWithTimeout(maxLoopDuration, sleepBetweenAttempts);
+
+        verifier.condition(() -> {
+                    Collection<Member> effectiveMemberList = getEffectiveMemberList(clientInstance);
+                    int activeConnections = getActiveConnections(clientInstance);
+                    return effectiveMemberList.size() == 1 && activeConnections == 1;
+                })
+                .onConditionPass(() -> {
+                    UUID newConnectedMemberUuid = getEffectiveMemberList(clientInstance)
+                            .iterator().next().getUuid();
+                    if (!newConnectedMemberUuid.equals(connectedMemberUuid)) {
+                        logger.info("Client is connected to a new member: "
+                                + newConnectedMemberUuid + " previously: " + connectedMemberUuid);
+                    }
+                    connectedMemberUuid = newConnectedMemberUuid;
+                })
+                .onConditionFail(
+                        () -> logger.warning("Assertion failed. Active connections: "
```

Comment:
Is it expected that it just log warning? What if it will fail over and over again during the whole test run - should it be considered as a correct behavior and test should finished as passed successfully? What's the condition where the test should fail?

Reply:
I changed this log to below version:
``` INFO: Waiting for expected connection... Currently, active connections: 0 effective members: [] ```
My assumption was that this can be useful in case of failure. Maybe client was connected to more than one members or maybe client lost connection at all (like in mentioned log from dynamic cluster). 
Condition will fail if won't be passed in a proper time.

Reply:
> My assumption was that this can be useful in case of failure.

But will it actually fail the test? I mean we can have full log of messages like this but if the test will pass nobody will check it.